File: frontend/src/microservices/traveler/objects/ListTravelersPlannerRequest.ts
// 本文件定义 traveler 模块的 `ListTravelersPlannerRequest`，作为列表请求参数并提供 JSON 编解码。

export type ListTravelersPlannerRequest = {
  actingUserId: string
  ownerUserId: string
}

export const listTravelersPlannerRequestFromJson = (json: string): ListTravelersPlannerRequest =>
  JSON.parse(json) as ListTravelersPlannerRequest

export const listTravelersPlannerRequestToJson = (value: ListTravelersPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/traveler/objects/ListTravelersPlannerRequest.scala
// 本文件定义 traveler 模块的列表请求 `ListTravelersPlannerRequest`，仅负责数据结构与 JSON codec。
package com.typesafe.travel.traveler.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ListTravelersPlannerRequest(
    actingUserId: String,
    ownerUserId: String
)

object ListTravelersPlannerRequest:
  given sourceEncoder: Encoder[ListTravelersPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[ListTravelersPlannerRequest] = deriveDecoder
